components/lib/inputotp/InputOtp.vue

```diff
@@ -128,49 +128,52 @@ export default {
         onBlur(event) {
             this.$emit('blur', event);
         },
+        isNumericKey(key) {
+            return !isNaN(Number(key)) || key === 'Backspace' || key === 'ArrowLeft' || key === 'ArrowRight';
+        },
         onKeyDown(event) {
-            switch (event.code) {
-                case 'ArrowLeft':
-                    this.moveToPrev(event);
-                    event.preventDefault();
-
-                    break;
-
-                case 'ArrowUp':
-                case 'ArrowDown':
-                    event.preventDefault();
-
-                    break;
-
-                case 'Backspace':
-                    if (event.target.value.length === 0) {
-                        this.moveToPrev(event);
-                        event.preventDefault();
-                    }
-
-                    break;
+            const { key, target } = event;
+            const index = target.dataset.index;
 
-                case 'ArrowRight':
-                    this.moveToNext(event);
-                    event.preventDefault();
+            if (this.integerOnly && !this.isNumericKey(key)) {
+                event.preventDefault();
+                return;
+            }
 
-                    break;
+            const movementKeys = {
+                ArrowLeft: () => this.moveToPrev(event),
+                ArrowRight: () => this.moveToNext(event)
+            };
 
-                default:
-                    if ((this.integerOnly && !((event.code.startsWith('Digit') || event.code.startsWith('Numpad')) && Number(event.code) >= 0 && Number(event.code) <= 9)) || (this.tokens.join('').length >= this.length && event.code !== 'Delete')) {
-                        event.preventDefault();
-                    }
+            const deletionConditions = {
+                Backspace: () => target.value === '' && target.selectionStart === 0,
+                Delete: () => target.value !== ''
+            };
 
-                    break;
+            if (movementKeys[key]) {
+                movementKeys[key]();
+                event.preventDefault();
+            } else if (deletionConditions[key] && deletionConditions[key]()) {
+                if (key === 'Delete') {
+                    target.value = '';
+                }
+                this.tokens[index] = '';
+                this.updateModel(event);
+                event.preventDefault();
             }
         },
+
         onPaste(event) {
-            let paste = event.clipboardData.getData('text');
+            let paste = event.clipboardData
+                .getData('text')
+                .split('')
+                .filter((char) => this.isNumericKey(char))
```

Comment:
Hmmm what happens if its a non integer only OTP and they paste "ABC" is this filtering out for only numeric keys?

Reply:
@melloware I removed it there, I realized that that part was not necessary due to the validation found below